### PR TITLE
[FLINK-17061][runtime] Unset TM total process/flink memory size for fine-grained resource management

### PR DESCRIPTION
**Pull request by xintongsong**

### CODE REVIEW

**Comment by xintongsong** [comment]
cc @KarmaGYZ

File: flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/active/ActiveResourceManagerFactory.java [hunk 1]
```diff
@@ -56,6 +59,33 @@ protected Configuration getEffectiveConfigurationForResourceManagerAndRuntimeSer
                 configuration, TaskManagerOptions.TOTAL_PROCESS_MEMORY);
     }
 
+    @Override
+    protected Configuration getEffectiveConfigurationForResourceManager(
```
Comment (KarmaGYZ) [inline flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/active/ActiveResourceManagerFactory.java:63]: We can add a test for it to verify:
- If the fine-grained resource management is enabled, those options should be removed.
- If the fine-grained resource management is not enabled, those options should be passed by.

**Review by KarmaGYZ — APPROVED** [review]
Thanks for the PR, @xintongsong . It looks really good. I only left one minor comment for the test.

**Comment by xintongsong** [comment]
Thanks for the review, @KarmaGYZ. Comments addressed.

**Comment by KarmaGYZ** [comment]
Thanks for addressing my comment. +1 for merging when the CI passed.

**Pull request closed without merging**